drivers/serial/uart_rcar.c

```diff
@@ -258,33 +289,312 @@ static int uart_rcar_init(const struct device *dev)
 		return ret;
 	}
 
-	return uart_rcar_configure(dev, &data->current_config);
+	ret = uart_rcar_configure(dev, &data->current_config);
+	if (ret != 0) {
+		return ret;
+	}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	config->irq_config_func(dev);
+#endif
+
+	return 0;
+}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+static bool uart_rcar_irq_is_enabled(const struct device *dev,
+				     uint32_t irq)
+{
+	const struct uart_rcar_cfg *config = DEV_UART_CFG(dev);
+	struct uart_rcar_data *data = DEV_UART_DATA(dev);
+
+	uint16_t reg_val;
+	k_spinlock_key_t key = k_spin_lock(&data->lock);
+
+	reg_val = !!(uart_rcar_read_16(config, SCSCR) & irq);
+
+	k_spin_unlock(&data->lock, key);
```

Comment:
If the hardware can read the register atomically, there should be no need to wrap it around a spin lock. I am not familiar with the target hardware. So if you think this locking is needed, I am okay with it. (ditto for two more read-only locking below)